src/EnergyPlus/SetPointManager.cc

```diff
@@ -3703,6 +3787,253 @@ void GetSetPointManagerInputData(EnergyPlusData &state, bool &ErrorsFound)
+                                                                 NumAlphas,
+                                                                 rNumericArgs,
+                                                                 NumNums,
+                                                                 IOStat,
+                                                                 lNumericFieldBlanks,
+                                                                 lAlphaFieldBlanks,
+                                                                 cAlphaFieldNames,
+                                                                 cNumericFieldNames);
+        UtilityRoutines::IsNameEmpty(state, cAlphaArgs(1), cCurrentModuleObject, ErrorsFound);
+
+        auto &setpointManager = state.dataSetPointManager->SystemNodeResetSetPtMgr(SetPtMgrNum);
+
+        setpointManager.Name = cAlphaArgs(1);
+        setpointManager.ctrlVarType = cAlphaArgs(2);
+
+        int typeNum = getEnumerationValue(SetPointManager::controlTypeNameUC, cAlphaArgs(2));
+        setpointManager.CtrlTypeMode = static_cast<SetPointManager::CtrlVarType>(typeNum);
+        if (setpointManager.CtrlTypeMode == SetPointManager::CtrlVarType::Invalid) {
+            // should not come here if idd type choice and key list is working
+            ShowSevereError(state, format("{}: {}=\"{}\", invalid field.", RoutineName, cCurrentModuleObject, cAlphaArgs(1)));
+            ShowContinueError(state, "..invalid " + cAlphaFieldNames(2) + "=\"" + cAlphaArgs(2) + "\".");
+            ShowContinueError(state, "..Valid values are \"Temperature\", \"MaximumTemperature\", or \"MinimumTemperature\".");
+            ErrorsFound = true;
+        }
+
+        setpointManager.SpAtLowRef = rNumericArgs(1);
+        setpointManager.SpAtHighRef = rNumericArgs(2);
+        setpointManager.LowRef = rNumericArgs(3);
+        setpointManager.HighRef = rNumericArgs(4);
+
+        setpointManager.RefNodeNum = GetOnlySingleNode(state,
+                                                       cAlphaArgs(3),
+                                                       ErrorsFound,
+                                                       DataLoopNode::ConnectionObjectType::SetpointManagerSystemNodeResetTemperature,
+                                                       cAlphaArgs(1),
+                                                       DataLoopNode::NodeFluidType::Blank,
+                                                       DataLoopNode::ConnectionType::Sensor,
+                                                       NodeInputManager::CompFluidStream::Primary,
+                                                       ObjectIsNotParent);
+
+        setpointManager.CtrlNodeListName = cAlphaArgs(4);
+        if (setpointManager.SpAtLowRef < setpointManager.SpAtHighRef) {
+            ShowWarningError(state, format("{}: {}=\"{}\", invalid setpoints.", RoutineName, cCurrentModuleObject, cAlphaArgs(1)));
```

Comment:
Is this really invalid? Will the linear interpolation fail with this?
I don't see a check here for `HighRef < LowRef`. That should generate a warning saying the average of the two setpoint values will be used.

Reply:
Thanks for the valid comment. We changed it to `HighRef < LowRef`, generating the warnings (similar to the ones in `SetpointManager:OutDoorAirReset`).